playwright/src/main/java/com/microsoft/playwright/junit/impl/PageExtension.java

```diff
@@ -0,0 +1,46 @@
+package com.microsoft.playwright.junit.impl;
+
+import com.microsoft.playwright.BrowserContext;
+import com.microsoft.playwright.Page;
+import org.junit.jupiter.api.extension.*;
+
+import static com.microsoft.playwright.junit.impl.ExtensionUtils.hasUsePlaywrightAnnotation;
+import static com.microsoft.playwright.junit.impl.ExtensionUtils.isClassHook;
+
+public class PageExtension implements ParameterResolver, AfterEachCallback {
+  private static final ThreadLocal<Page> threadLocalPage = new ThreadLocal<>();
+  @Override
+  public void afterEach(ExtensionContext extensionContext) {
+    Page page = threadLocalPage.get();
+    threadLocalPage.remove();
+    if(page != null) {
+      page.close();
+    }
+  }
+
+  @Override
+  public boolean supportsParameter(ParameterContext parameterContext, ExtensionContext extensionContext) throws ParameterResolutionException {
```

Comment:
`supportsParameter/resolveParameter` logic is very similar across the extensions, consider extracting this in a common superclass parametrized by the parameter class.